storage/tokudb/PerconaFT/CMakeLists.txt

```diff
@@ -52,6 +52,10 @@ include(TokuThirdParty)
 set(TOKU_CMAKE_SCRIPT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 include(TokuMergeLibs)
 
+if (NOT HAVE_O_DIRECT AND NOT HAVE_F_NOCACHE)
+   RETURN()
+endif()
+
```

Comment:
This wouldn't do. Try to build. You disable PerconaFT library, but not the tokudb plugin itself. It'll fail to build because some of its dependencies won't be built.

The check is fine, but you need it in storage/tokudb/CMakeLists.txt